File: workflow-manager/src/main/java/fii/workflow/manager/domain/Execution.java
package fii.workflow.manager.domain;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
public class Execution {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long stepNumber;
    private String message;
    private String status;

    private LocalDateTime startTime;
    private LocalDateTime endTime;

    @ManyToOne
    @JoinColumn(name = "workflow_id")
    private Workflow workflow;

    @OneToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;
}

File: workflow-manager-frontend/app/src/main/java/com/example/workflow_manager_frontend/domain/Execution.kt
package com.example.workflow_manager_frontend.domain

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Execution(
    val id: Int,
    val startTime: Long?,
    val endTime: Long?,
    val status: String,
    val message: String?,
    val stepNumber: Int,
    val workflowId: Int,
    val customer: Customer?
) : Parcelable